hls-plugin-api/src/Ide/Plugin/Config.hs

```diff
@@ -79,7 +90,18 @@ parseConfig defValue = A.withObject "Config" $ \v -> do
         <*> (o .:? "checkProject" <|> v .:? "checkProject") .!= checkProject defValue
         <*> o .:? "formattingProvider"                      .!= formattingProvider defValue
         <*> o .:? "maxCompletions"                          .!= maxCompletions defValue
-        <*> o .:? "plugin"                                  .!= plugins defValue
+        <*> explicitParseFieldMaybe (parsePlugins $ plugins defValue) o "plugin" .!= plugins defValue
+
+parsePlugins :: Map.Map T.Text PluginConfig -> Value -> A.Parser (Map.Map T.Text PluginConfig)
+parsePlugins defValue = A.withObject "Config.plugins" $ \o -> do
+  let -- parseOne :: Key -> Value -> A.Parser (T.Text, PluginConfig)
```

Comment:
any reason to not just leave the signature in?

Reply:
it would require CPP, since `Key` is only defined in aeson 2.0